Add guide for deploying MI integrations on k8s using helm chart. (#995)
wso2/docs-mi: en/docs/install-and-setup/setup/deployment/deploying-micro-integrator-with-helm.md

                    tag: "<custom_image_tag>"
        ```

    - You can update [other configurations](https://github.com/wso2/helm-mi/blob/main/mi/EXAMPLES.md) as needed.

3. Save the `values.yaml` file.

## Step 3 - Deploy the Micro Integrator

Once you have configured your Helm resources locally, follow the instructions below to deploy the Micro Integrator.

1. Open a terminal and navigate to the `<HELM_HOME>/mi` folder.
2. Run the command appropriate for your Helm version.

    !!! Tip
        Be sure to replace `NAMESPACE` with the Kubernetes namespace where your resources are deployed.

    - Using **Helm v3**:
        
        ```bash
        helm install <RELEASE_NAME> ./ -f values.yaml -n <NAMESPACE> --create-namespace
        ```

!!! Note

>     For a pre configured, easy setup for local deployment, use the `values_local.yaml` file.

Reviewer: ```suggestion
    For a pre-configured, easy setup for local deployment, use the `values_local.yaml` file.
```

    ```bash
    curl https://mi.wso2.com/healthz -k
    ```

> ## Deploy the Micro Integrator with an Integration

Reviewer: ```suggestion
## Deploy the Micro Integrator with an integration
```

> To deploy the Micro Integrator with an integration (e.g., the HelloDocker sample), follow the instructions below. This section assumes you have already built a custom Docker image containing your integration, as outlined in the [HelloDocker sample guide]({{base_path}}/learn/samples/docker-sample/) and the [guide for generating Docker images]({{base_path}}/develop/generate-docker-image/).

Reviewer: ```suggestion
To deploy the Micro Integrator with an integration (For example, HelloDocker sample), follow the instructions below. This section assumes you have already built a custom Docker image containing your integration, as outlined in the [HelloDocker sample guide]({{base_path}}/learn/samples/docker-sample/) and the [guide for generating Docker images]({{base_path}}/develop/generate-docker-image/).
```

> To deploy the Micro Integrator with an integration (For example, HelloDocker sample), follow the instructions below. This section assumes you have already built a custom Docker image containing your integration, as outlined in the [HelloDocker sample guide]({{base_path}}/learn/samples/docker-sample/) and the [guide for generating Docker images]({{base_path}}/develop/generate-docker-image/).

Reviewer: ```suggestion
To deploy the Micro Integrator with an integration (For example, `HelloDocker` sample), follow the instructions below. This section assumes you have already built a custom Docker image containing your integration, as outlined in the [HelloDocker sample guide]({{base_path}}/learn/samples/docker-sample/) and the [guide for generating Docker images]({{base_path}}/develop/generate-docker-image/).
```

> ### Step 1 - Build and Push Your Custom Docker Image

Reviewer: ```suggestion
### Step 1 - Build and push your custom Docker image
```

> 1. Build the Docker image containing your integration (e.g., HelloDocker) as per the [HelloDocker sample guide]({{base_path}}/learn/samples/docker-sample/).

Reviewer: ```suggestion
1. Build the Docker image containing your integration (For example, `HelloDocker`) as per the [HelloDocker sample guide]({{base_path}}/learn/samples/docker-sample/).
```

   
2. Once built, tag and push the image to a Docker registry or make it available locally for use in your Kubernetes cluster.

    - If you are using a local registry, no need to push the image to a remote repository.

> ### Step 2 - Update the `values.yaml` File for Deployment

Reviewer: ```suggestion
### Step 2 - Update the `values.yaml` file for deployment
```

1. Open the `values.yaml` or `values_local.yaml` file in the `<HELM_HOME>/mi` directory.

2. Update the image repository and tag to point to the custom image you built (in this case, `hellodocker` as the repository and `1.0.0` as the tag). If you are using a local image, remove the `containerRegistry` entry.

    ```yaml
    # -- Container registry (leave this empty for local image)
    containerRegistry: ""

    wso2:
      deployment:
        image:
          repository: "hellodocker"  # Use your custom image repository
          tag: "1.0.0"  # Use the tag of your custom image
    ```

> 3. **If you are using a private Docker registry**, you will also need to add the `imagePullSecrets` parameter to your `values.yaml` file to authenticate against the registry. Set `imagePullSecrets` to the name of the Kubernetes secret that stores your Docker registry credentials.

Reviewer: ```suggestion
3. If you are using a private Docker registry, you will also need to add the `imagePullSecrets` parameter to your `values.yaml` file to authenticate against the registry. Set `imagePullSecrets` to the name of the Kubernetes secret that stores your Docker registry credentials.
```

    ```yaml
    wso2:
      deployment:
        imagePullSecrets: "<your_image_pull_secret>"  # Specify your image pull secret here
    ```

4. Save the changes to the `values.yaml` file. 

This configuration ensures that Kubernetes pulls the custom Docker image from your private registry using the specified credentials.

> ### Step 3 - Deploy the Micro Integrator with the Custom Integration

Reviewer: ```suggestion
### Step 3 - Deploy the Micro Integrator with the custom integration
```

For a local deployment, use the `values_local.yaml` file, which is specifically configured for local setups without cloud-specific configurations.

1. Navigate to the `<HELM_HOME>/mi` directory in your terminal.
2. Run the Helm command using the `values_local.yaml` file for local deployment. Use the appropriate command based on your Helm version:

    - For **Helm v3**:

      ```bash
      helm install <RELEASE_NAME> ./ -f values_local.yaml -n <NAMESPACE> --create-namespace
      ```

    Make sure to specify the correct Kubernetes namespace (`<NAMESPACE>`) for the deployment.

> 3. Test the custom integration by running the following `curl` command:

Reviewer: ```suggestion
3. Test the custom integration by running the following curl command:
```
